### DISCUSSION TITLE
Bind function in QueryBinder

### DISCUSSION THREAD

**Discussion by Adu2120** [iTwin/itwinjs-core#4606] [Q&A] [answered]
Can you please describe the indexOrName parameter of all bind function in QueryBinder?

Also please describe the respective index of the name.

**Comment by khanaffan** [comment, marked as answer]
I will update the documentation. Though basically it means position or name of the parameter in query. They cannot be mixed. Either all query parameters must be index base or name base.

### Parameter by index
```sql
    SELECT a, v FROM test.Foo WHERE a=? AND b=?
```

The first (?) = 1 and the second (?) is 2.  The parameter index starts with one and not 0.

### Parameter by name
```sql
    SELECT a, v FROM test.Foo WHERE a=:a_v AND b=:b_v
```

`a_v` and `b_v` are the name of parameters.

**Comment by Adu2120** [comment]
Okay. Thank you
